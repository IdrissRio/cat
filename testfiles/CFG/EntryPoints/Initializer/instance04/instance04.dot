digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "FieldDecl[28]"; "InstanceInitializer[36]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Modifiers[29]"; "PrimitiveTypeAccess[31]"; "List[32]"; "Block[37]"}
{ rank = same; "List[30]"; "FieldDeclarator[33]"; "List[38]"}
{ rank = same; "List[34]"; "Opt[35]"; "ExprStmt[39]"}
{ rank = same; "Dot[40]"}
{ rank = same; "Dot[41]"; "MethodAccess[44]"}
{ rank = same; "TypeAccess[42]"; "VarAccess[43]"; "List[46]"; "List[48]"; "List[49]"; "UncheckedExceptions[45]"}
{ rank = same; "StringLiteral[47]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < instance04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDecl[28]" [label="FieldDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[29]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[31]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[32]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"FieldDeclarator[33]" [label="FieldDeclarator\n < j >", fillcolor=white   style=filled  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[35]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitializer[36]" [label="InstanceInitializer\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[37]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[39]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[40]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[41]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[42]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[43]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[44]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[46]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[47]" [label="StringLiteral\n < test >", fillcolor=white   style=filled  ]
"List[48]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[45]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"Modifiers[29]" -> "List[30]" [style=dashed, color=gray]
"FieldDecl[28]" -> "Modifiers[29]" [style=dashed, color=gray]
"FieldDecl[28]" -> "PrimitiveTypeAccess[31]" [style=dashed, color=gray]
"FieldDeclarator[33]" -> "List[34]" [style=dashed, color=gray]
"FieldDeclarator[33]" -> "Opt[35]" [style=dashed, color=gray]
"List[32]" -> "FieldDeclarator[33]" [style=dashed, color=gray]
"FieldDecl[28]" -> "List[32]" [style=dashed, color=gray]
"List[27]" -> "FieldDecl[28]" [style=dashed, color=gray]
"Dot[41]" -> "TypeAccess[42]" [style=dashed, color=gray]
"Dot[41]" -> "VarAccess[43]" [style=dashed, color=gray]
"Dot[40]" -> "Dot[41]" [style=dashed, color=gray]
"List[46]" -> "StringLiteral[47]" [style=dashed, color=gray]
"MethodAccess[44]" -> "List[46]" [style=dashed, color=gray]
"MethodAccess[44]" -> "List[48]" [style=dashed, color=gray]
"MethodAccess[44]" -> "List[49]" [style=dashed, color=gray]
"MethodAccess[44]" -> "UncheckedExceptions[45]" [style=dashed, color=gray]
"Dot[40]" -> "MethodAccess[44]" [style=dashed, color=gray]
"ExprStmt[39]" -> "Dot[40]" [style=dashed, color=gray]
"List[38]" -> "ExprStmt[39]" [style=dashed, color=gray]
"Block[37]" -> "List[38]" [style=dashed, color=gray]
"InstanceInitializer[36]" -> "Block[37]" [style=dashed, color=gray]
"List[27]" -> "InstanceInitializer[36]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "MethodAccess[44]" [color=red, constraint=false]
"MethodAccess[44]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[44]" -> "StringLiteral[47]" [color=red, constraint=false]
"StringLiteral[47]" -> "MethodAccess[44]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[47]" -> "VarAccess[43]" [color=red, constraint=false]
"VarAccess[43]" -> "StringLiteral[47]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[43]" -> "TypeAccess[42]" [color=red, constraint=false]
"TypeAccess[42]" -> "VarAccess[43]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[42]" -> "FieldDeclarator[33]" [color=red, constraint=false]
"FieldDeclarator[33]" -> "TypeAccess[42]" [color=blue, constraint=false, xlabel=" "] 
"FieldDeclarator[33]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "FieldDeclarator[33]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"FieldDecl[28]" -> "InstanceInitializer[36]" [style=invis]
"Modifiers[29]" -> "PrimitiveTypeAccess[31]" -> "List[32]" [style=invis]
"List[34]" -> "Opt[35]" [style=invis]
"Dot[41]" -> "MethodAccess[44]" [style=invis]
"TypeAccess[42]" -> "VarAccess[43]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
40", fillcolor=1]
Edges [label="Total Nr. Edges
53", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
8", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
7", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}