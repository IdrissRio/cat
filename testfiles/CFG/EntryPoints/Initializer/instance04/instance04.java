public class instance04 {
  int j;
  { System.out.println("test"); }
}