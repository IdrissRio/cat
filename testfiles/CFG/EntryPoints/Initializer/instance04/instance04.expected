NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [FieldDeclarator[33]]
NodeName: FieldDeclarator[33] Successors: [TypeAccess[42]]
NodeName: TypeAccess[42] Successors: [VarAccess[43]]
NodeName: VarAccess[43] Successors: [StringLiteral[47]]
NodeName: StringLiteral[47] Successors: [MethodAccess[44]]
NodeName: MethodAccess[44] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [MethodAccess[44]]
NodeName: MethodAccess[44] Predecessor: [StringLiteral[47]]
NodeName: StringLiteral[47] Predecessor: [VarAccess[43]]
NodeName: VarAccess[43] Predecessor: [TypeAccess[42]]
NodeName: TypeAccess[42] Predecessor: [FieldDeclarator[33]]
NodeName: FieldDeclarator[33] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
